lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2214,9 +2242,23 @@ class SqlParser(private val ion: IonSystem) : Parser {
         return ParseNode(ARG_LIST, null, items, rem)
     }
 
+    /**
```

Comment:
There are other tokens that should only exist at top level right? Why is `EXEC` different and requires a separate check?

Reply:
The other tokens that should only exist at the top level (i.e. DML and DDL ops) weren't properly checked in the parser (created an issue #354 to add these checks in the future). I will rename the `checkForUnexpectedExec()` to something more general for all tokens that should only exist in the top level.